cashu/mint/ledger.py

```diff
@@ -25,6 +27,9 @@
 
 
 class Ledger:
+    locks: Dict[str, LockBase] = {}  # holds multiprocessing locks
+    proofs_pending_lock: LockBase = mp.Lock()  # holds locks for proofs_pending database

```

Comment:
Do you know if there's precedent for using `multiprocessing.Lock` with async? Concern here would be the usual that if it's blocking it'll cause similar async problems/thread-lockups to the blocking requests.

There's also `asyncio.Lock` https://docs.python.org/3/library/asyncio-sync.html

Reply:
Thanks for the review! I chose to not use `asyncio.Lock` (I used it before this PR) because eventually we want to be able to run the nutshell mint on multiple workers and only the multiprocessing lock allows you do lock across different threads. See here: https://stackoverflow.com/questions/18213619/sharing-a-lock-between-gunicorn-workers

Reply:
I don't know if there is any interference between the two though, I will give it a test and see if the lock actually works.

Reply:
If it does turn out to be blocking, you can always run in an executor https://docs.python.org/3/library/asyncio-dev.html#running-blocking-code -- adds a bit of overhead but would work

Reply:
It was indeed blocking. Once two threads ran into the lock, the whole thing stops. I reverted to asyncio locks for now but this won't be usable if we use multiprocessing in the future.